lib/extensions/ar_virtual.rb

```diff
@@ -141,31 +141,33 @@ def define_delegate(method_name, method, to: nil, allow_nil: nil, default: nil)
       # whereas conceptually, from the user point of view, the delegator should
       # be doing one call.
       if allow_nil
-        method_def = [
-          "def #{method_name}(#{definition})",
-          "_ = #{to}",
-          "if !_.nil? || nil.respond_to?(:#{method})",
-          "  _.#{method}(#{definition})",
-          "end#{default}",
-          "end"
-        ].join ';'
+        method_def = <<-METHOD
+          def #{method_name}(#{definition})
+            return self[:#{method_name}] if has_attribute?(:#{method_name})
```

Comment:
this is the only line change.
the other parts of the diff were the request to heredoc this